sympy/solvers/simplex.py

```diff
@@ -0,0 +1,821 @@
+    is returned if possible:
+
+    >>> lp(max, f, c + [x3 < oo])  # x3 can have any value
+    (5, {x1: 1, x2: 0, x3: -2, x4: 2})
+
+    Symbols not in the objective function will not be reported
+    and are treated as 0:
+
+    >>> lp(min, x, [y + x >= -3])  # same as lp(min, x, [x >= -3])
+    (-3, {x: -3})
+    """
+    A, B, C, D, r, xx, aux = _lp_matrices(f, constr)
+
+    how = str(min_max).lower()
+    if 'max' in how:
+        # _simplex minimizes for Ax <= B so we
+        # have to change the sign of the function
+        # and negate the optimal value returned
+        _o, p, d = _simplex(A, B, -C, -D)
+        o = -_o
+    elif 'min' in how:
+        o, p, d = _simplex(A, B, C, D)
+    else:
+        raise ValueError('expecting min or max')
+
+    # restore original variables and remove aux from p
+    p = dict(zip(xx, p))
+    if r:  # p has original symbols and auxilliary symbols
+        # if r has x: x - z1 use values from p to update
+        r = {k: v.xreplace(p) for k, v in r.items()}
+        # then use the actual value of x (= x - z1) in p
+        p.update(r)
+        # and keep only non-aux variables
+        p = {k: v for k, v in p.items() if k not in aux}
+
+    # not returning dual since there may be extra constraints
+    # when a variable has finite bounds
+    return o, p
+
+
+def _lp_args(min_max, *args):
+    """return standard input for lp from how and args. The
+    method ``how`` is interpreted as being either min or max,
+    and args (up to 4) are interpreted and returned as an
+    object function and list of appropriate inequalities.
+
+    1 arg: Matrix/list
+    2 args: Expr, list of constraints
+    3 args: 3 lists/matrices: A, B, C, D=0
+    4 args: 4 lists/matrices: A, B, C, D
+
+    return: ('max' or 'min', objective, constraints)
+    """
+    how = str(min_max).lower()
+    if 'max' in how:
+        how = 'max'
+    elif 'min' in how:
+        how = 'min'
+    else:
+        raise ValueError('expecting min or max for `how`')
+    if not args or len(args) > 4:
+        raise ValueError('expecting 1 - 4 args')
+    LP = [sympify(i) for i in args]
+    if len(LP) > 2 and all(isinstance(i, (list, MatrixBase)) for i in LP):
+        if len(LP) == 3:
+            LP.append([0])
+        a, b, c, d = [Matrix(i) for i in LP]
+        LP = [Matrix([[a, b], [c, d]])]
+    if len(LP) == 1:
+        m = Matrix(LP[0])
+        if how == 'min':
+            p = primal_dual(m)[0]
+        elif how == 'max':
+            p = primal_dual(m.T)[1]
+        return (how,) + p
+    if len(LP) == 2:
+        LP.append([])
+    f, ineq, unbound = LP
+    if not isinstance(f, Expr):
+        raise ValueError('expecting Expr')
+    if not isinstance(ineq, list):
+        raise ValueError('expecting list of inequalities')
+    if not isinstance(unbound, list):
+        raise ValueError('expecting list of unbound symbols')
+    return how, f, ineq
+
+
+def lpmin(*args):
```

Comment:
Instead of *args, the parameters could be 

objective, constraints 

where 

objective : either an expression or Matrix([c, d])

constraints : either a list of  constraints expressed using Ge, Le or Eq or Matrix([[A, b]])

Reply:
See below: there can be 1 (matrix), 2 (f, constr), 3 or 4 args (individual matrix components).